extension-style/src/main/java/com/mapbox/maps/extension/style/sources/generated/GeoJsonSource.kt

```diff
@@ -471,7 +471,12 @@ class GeoJsonSource(builder: Builder) : Source(builder.sourceId) {
    * @param value the feature
    */
   fun feature(value: Feature) = apply {
-    setProperty(PropertyValue("data", value.toValue()))
+    workerHandler.post {
```

Comment:
By adding this async behavior, will this assertion pass?
```
val expected = ...
source.data(expected)
val resulting = source.data
assertEquals(expected, resulting)
```

I imagine it won't. Would it pass in the current main branch, is the round-tripping currently synchronous in this way? 

Should the getter have a callback instead of being a property, if the setter is asynchronous?

Reply:
To me above makes sense but in this case it's maybe not needed. gl-native doesn't allow to get back inline geojson, they don't keep a reference to it (see mapbox-gl-native-internal#522).

Reply:
> Would love to see the UX for animating a GeoJsonSource geometry. We used to have an example with cars driving around a map https://github.com/mapbox/mapbox-gl-native-android/blob/main/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/activity/annotation/AnimatedSymbolLayerActivity.java but feel free to create something more useful from scratch.
> 
> Are there any other metric data points that we capture to quantify the performance improvements? It's clear that unblocking the main thread, UX wise, is a lot better but quantifying it with numbers would be great for long term metric tracking.

I think that basically we could simply add benchmark with sync / async loading geojson in our metrics repo and check out results. Or are you talking about something else?